crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1796,6 +1822,13 @@ pub struct BytesLiteralType<'db> {
     value: Box<[u8]>,
 }
 
+#[salsa::interned]
+pub struct SliceLiteralType<'db> {
+    start: Option<i64>,
+    stop: Option<i64>,
+    step: Option<i64>,
+}
```

Comment:
Another option I considered was to inline this within `Type`. This would increase the size of `Type`. Unless we think it's okay to restrict indices for literal slice types to `i32`. Which might be fine. It seems unlikely that someone has a >2 GiB literal string inside their Python source code *and still* cares about inferring a static type for something like `str_literal[4_294_967_297:]`.

Reply:
I think using an `i32` is probably okay. Ruff/Red Knot also only supports files `len(source) = u32::MAX`

Reply:
Switched to `i32` now as this also solves some problems on 32 bit platforms (see other comment). It's still `salsa::interned` for now, as three `Option<i32>`s (24 byte) are still larger than `Type` at the moment (16 byte).